internal/codegen/tpl_file.go

```diff
@@ -122,6 +123,13 @@ func (f *TplFile) Static() (out string, err error) {
 //
 //	{{- file.Once }}
 func (f *TplFile) Once() (out string, err error) {
+	// if the file exists at all, skip it
+	if _, err := os.Stat(f.f.path); err == nil {
+		f.log.With("template", f.t.Path, "path", f.f.path).
+			Debug("Skipping once file because it already exists on FS")
+		return f.Skip("Once file, output already exists")
+	}
```

Comment:
Can you copy the logic used here: https://github.com/rgst-io/stencil/blob/main/internal/codegen/tpl_stencil.go#L229-L233?

It helps prevent out-of-bounds file reading (on accident). It's not 10000% robust, but I've been keeping stuff within `osfs` when I can